src/app/SecurityPanel/CreateJmxCredentialModal.tsx

```diff
@@ -0,0 +1,96 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import { Form, FormGroup, Modal, ModalVariant, ValidatedOptions } from '@patternfly/react-core';
+import { JmxAuthForm } from '@app/AppLayout/JmxAuthForm';
+import { TargetSelect } from '@app/TargetSelect/TargetSelect';
+import { NO_TARGET } from '@app/Shared/Services/Target.service';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { first } from 'rxjs';
+
+export interface CreateJmxCredentialModalProps {
+  visible: boolean;
+  onClose: () => void;
+}
+
+export const CreateJmxCredentialModal: React.FunctionComponent<CreateJmxCredentialModalProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const [validTarget, setValidTarget] = React.useState(ValidatedOptions.default);
+
+  const onSave = React.useCallback((username: string, password: string) => {
+    let isValid;
+    context.target.target().subscribe((t) => {
+      isValid = t == NO_TARGET ? ValidatedOptions.error : ValidatedOptions.success;
+      setValidTarget(isValid);
+    });
+
+    if (isValid == ValidatedOptions.success) {
+      context.api.postTargetCredentials(username, password)
+      .pipe(first())
+      .subscribe();
+      props.onClose();
+    }
+  }, [props.onClose, context, context.target, context.api, validTarget, setValidTarget]);
+
+  return (
+    <Modal
+      isOpen={props.visible}
+      variant={ModalVariant.large}
+      showClose={true}
+      onClose={props.onClose}
+      title="Store JMX Credentials"
+      description="Creates stored credentials for a given target. 
+        If a Target JVM requires JMX authentication, Cryostat will use stored credentials 
+        when attempting to open JMX connections to the target."
+    >
+      <Form>
+        <FormGroup
+          label="Target"
+          isRequired
+          fieldId="target-select"
+          helperTextInvalid="Select a target"
+          validated={validTarget}
+        >
+          <TargetSelect />
+        </FormGroup>
+      </Form>
+      <br/>
+      <JmxAuthForm onSave={onSave} onDismiss={props.onClose} />
```

Comment:
The form focus on this modal starts at `Username` instead of `Target`, which means someone using the keyboard needs to tab through the entire modal before the focus lands on the `TargetSelect` dropdown. Adding/removing the `Form` component doesn't change the behaviour. Any ideas?

Reply:
I think that's kind of okay actually, since the `TargetSelect` will have automatically selected whichever target the user was looking at last.

Users can use shift-tab to move upward from the username field to the target selection dropdown too if they really want to change it.